MichaelKintscher/qualitative-coding-app
# Edit and Remove Buttons from User Settings Page

This patch moves the ability to edit global button definitions from the button panel to the user settings page. This patch also adds functionality to be able to remove a single button definition or clear the entire list. This patch also ensures a button definition cannot be saved with the same name as a previous definition.

Testing Steps:
  1. Create a new session
  2. Click "Settings" in the top left and open the user settings page
  3. Ensure your Button Definition Settings is empty (If not, click "Remove Definition", click "Clear all Button Definitions", and click "Yes". Skip steps 19-20)
  4. Exit the Settings page
  5. Click "+" and create a new button with the name "New Button" and click "Yes" when prompted to save the button definition
  6. Open the Settings page
  7. Ensure the New Button definition has populated in the list
  8. Exit the Settings page
  9. Repeat step 5 with a new hotkey
  10. Open the Settings page
  11. Ensure a second "New Button" definition has NOT populated in the list
  12. Click "Remove Definition"
  13. Enter "New Button" into the field and click "Remove Button Definition"
  14. Ensure the button is deleted from the list immediately
  15. Exit the Settings Page
  16. Create and save two new buttons with different names and hotkeys
  17. Open the settings page and ensure your definitions were saved
  18. Click "Remove Definition"
  19. Click "Clear all Button Definitions" and click "Yes" when prompted
  20. Ensure all button definitions are removed from the list immediately

Type: New Feature
